components/AvalancheForecastZonePolygon.tsx

```diff
@@ -85,59 +65,27 @@ export const updateRegionToContain = (previous: Region, coordinates: LatLng[]):
 };
 
 export interface AvalancheForecastZonePolygonProps {
-  feature: Feature;
-  date: string;
+  zone: MapViewZone;
   setRegion: React.Dispatch<React.SetStateAction<Region>>;
+  setSelectedZone: (zone: MapViewZone) => void;
 }
 
-export const AvalancheForecastZonePolygon: React.FunctionComponent<AvalancheForecastZonePolygonProps> = ({feature, date, setRegion}: AvalancheForecastZonePolygonProps) => {
-  const coordinates: LatLng[] = coordinateList(feature.geometry).map(toLatLng);
+export const AvalancheForecastZonePolygon: React.FunctionComponent<AvalancheForecastZonePolygonProps> = ({zone, setRegion, setSelectedZone}: AvalancheForecastZonePolygonProps) => {
+  const coordinates: LatLng[] = coordinateList(zone.geometry).map(toLatLng);
   React.useEffect(() => {
     setRegion((previous: Region) => updateRegionToContain(previous, coordinates));
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-  const forecastDate: Date = parseISO(date);
-  const navigation = useNavigation<HomeStackNavigationProps>();
-  const {isLoading, isError, data: forecast, error} = useAvalancheForecastFragment(feature.properties.center_id, feature.id, forecastDate);
-  if (isLoading) {
-    return (
-      <NotificationMarker geometry={feature.geometry}>
-        <ActivityIndicator />
-      </NotificationMarker>
-    );
-  }
-  if (isError) {
-    return (
-      <NotificationMarker geometry={feature.geometry}>
-        <View>
-          <Text>{`Could not fetch forecast for ${feature.properties.name}: ${error?.message}.`}</Text>
-        </View>
-      </NotificationMarker>
-    );
-  }
-
-  let dangerLevel: DangerLevel = DangerLevel.None;
-  if (forecast) {
-    const currentDanger: AvalancheDangerForecast | undefined = forecast.danger.find(item => item.valid_day === ForecastPeriod.Current);
-    if (currentDanger) {
-      dangerLevel = Math.max(currentDanger.lower, currentDanger.middle, currentDanger.upper);
-    }
-  }
-
```

Comment:
Polygons don't need to fetch their own data, display loading errors, etc. The parent component (AFZoneMap) has already done the hard work.